src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
@lzybkr this discussion thread was auto-collapsed in the main PR page, but you can read it at https://github.com/PowerShell/PowerShell/pull/2421/files/70c30384c74d9f6392b13cc9d04baf9eb5f8c364